### PR TITLE
L2ExecutePayloadProof Hint Type

### PR DESCRIPTION
**Pull request by zobront**
In order to use Kona with reth, we need to get around the `debug_dbGet` calls.

Reth has recently implemented the `debug_executePayload` RPC method. This allows us to bulk get the entire witness (codes, states, and keys) for executing a given set of payloads against a prior state.

To minimally modify Kona to support this, this PR:
- Introduces the `L2ExecutePayloadProof` hint type
- Implements this hint in the fetcher to get all the data and add it all to the KV store

This will allow a program that uses Kona as a library to perform a bulk fetch using this hint type before execution. If it succeeds, all L2 state related hints (except the final output root calculation, which doesn't use `debug_dbGet`) will be supplied with an existing KV store value in `get_preimage()` and will never need to call `prefetch()`.

When Kona is upgraded to use this RPC method in the future, it can be integrated more deeply into the existing MPT library (and eventually, when fully trusted, can replace a lot of it). For now, this PR will give users the option to manually use this method to avoid further RPC calls.

### CODE REVIEW

File: bin/host/src/fetcher/mod.rs
Comment (clabby) [inline bin/host/src/fetcher/mod.rs:528]: This is fine. The values will be de-duplicated in favor of most recent insertion.

**Review by clabby — APPROVED** [review]
lgtm!

File: bin/host/src/fetcher/mod.rs
Comment (clabby) [inline bin/host/src/fetcher/mod.rs:521]: has this been tested? Curious if it's expecting a `Vec<u8>` (json enc: `[0, 1, 2, 3, ...]`) instead of a `String`ified JSON (enc: `"{\"a\": 1, ...}"` for the req. Over in `reth` it's just deserializing the `Attributes` generic verbatim, which should be `OptimismPayloadAttributesV3` under the hood, but just sanity checking.

Comment (zobront) [reply bin/host/src/fetcher/mod.rs:521]: Waiting on some help from reth team to get tests working on this endpoint. Will test everything before merging — just wanted to get your feedback early in case there were more changes to make.

Thank you!

**Comment by tynes** [comment]
Can we add the `L2ExecutePayloadProof` to the specs?

**Comment by zobront** [comment]
@clabby I ran a test and it works as expected. Didn't leave the test in there because there aren't any public RPCs implementing this method yet. Let me know if there's another way you'd like to include the test - otherwise, this is good to merge.

**Review by refcell — APPROVED** [review]
LGTM pending @clabby's approval

**Review by clabby — APPROVED** [review]

**Pull request merged**
